py/htm/encoders/eye.py

```diff
@@ -0,0 +1,588 @@
+        - Adds 5 dots to the center of the image to show where the fovea is.
+
+        Returns an RGB image.
+        """
+        if roi is None:
+            roi = self.roi
+
+        # Show the ROI, first rotate it like the eye is rotated.
+        angle = self.orientation * 360 / (2 * math.pi)
+        roi = self.roi[:,:,::-1]
+        rows, cols, color_depth = roi.shape
+        M   = cv2.getRotationMatrix2D((cols / 2, rows / 2), angle, 1)
+        roi = cv2.warpAffine(roi, M, (cols,rows))
+
+        # Mask out areas the eye can't see by drawing a circle boarder.
+        center = int(roi.shape[0] / 2)
+        circle_mask = np.zeros(roi.shape, dtype=np.uint8)
+        cv2.circle(circle_mask, (center, center), center, thickness = -1, color=(255,255,255))
+        roi = np.minimum(roi, circle_mask)
+
+        # Invert 5 pixels in the center to show where the fovea is located.
+        roi[center, center]     = np.full(3, 255) - roi[center, center]
+        roi[center+2, center+2] = np.full(3, 255) - roi[center+2, center+2]
+        roi[center-2, center+2] = np.full(3, 255) - roi[center-2, center+2]
+        roi[center-2, center-2] = np.full(3, 255) - roi[center-2, center-2]
+        roi[center+2, center-2] = np.full(3, 255) - roi[center+2, center-2]
+        return roi
+
+    def show_view(self, window_name='Eye'):
+        if False:
+            print("Sparsity %g"%(len(self.output_sdr) / self.output_sdr.size))
+            parvo = self.output_sdr.dense[:,:,0]
+            magno = self.output_sdr.dense[:,:,1]
+            print("Parvo Sparsity %g"%(np.count_nonzero(parvo) / np.product(parvo.shape)))
+            print("Magno Sparsity %g"%(np.count_nonzero(magno) / np.product(magno.shape)))
+        roi = self.make_roi_pretty()
+        cv2.imshow('Region Of Interest', roi)
+        cv2.imshow('Parvocellular', self.parvo[:,:,::-1])
+        cv2.imshow('Magnocellular', self.magno)
+        cv2.waitKey(1)
+
+    def input_space_sample_points(self, npoints):
+        """
+        Returns a sampling of coordinates which the eye is currently looking at.
+        Use the result to determine the actual label of the image in the area
+        where the eye is looking.
+        """
+        # Find the retina's radius in the image.
+        r = int(round(self.scale * self.retina_diameter / 2))
+        # Shrink the retina's radius so that sample points are nearer the fovea.
+        # Also shrink radius B/C this does not account for the diagonal
+        # distance, just the manhattan distance.
+        r = r * 2/3
+        # Generate points.
+        coords = np.random.random_integers(-r, r, size=(npoints, 2))
+        # Add this position offset.
+        coords += np.array(np.rint(self.position), dtype=np.int).reshape(1, 2)
+        return coords
+
+    def reset(self):
+        self.retina.clearBuffers()
+
+
+class EyeSensorSampler:
+    """
+    Samples eyesensor.rgb, the eye's view.
+
+    Attribute samples is list of RGB numpy arrays.
```

Comment:
make this accept a jpg/png image

Reply:
This should accept anything the python imaging library `PIL` accepts.